### ISSUE TITLE
Implement Sign Typed Data V3 for signing relayed call data

### ISSUE THREAD

**Issue by peterpolman** [thxprotocol/monorepo-legacy#9]

**Comment by peterpolman** [comment]
https://docs.metamask.io/guide/signing-data.html#signing-data-with-metamask

**Comment by peterpolman** [comment]
https://medium.com/metamask/eip712-is-coming-what-to-expect-and-how-to-use-it-bb92fd1a7a26

**Comment by peterpolman** [comment]
This will also require adjusting the RelayHubFacet to support verifying the typed data structs.

**Comment by peterpolman** [comment]
- [ ] Test Promotions Metamask
- [ ] Test Swaps Metamask
- [ ] Test Withdrawal Metamask
- [ ] Add event listeners for Metamask onChain onAccount changing
- [ ] Test Payments for both Metamask and default

**Comment by peterpolman** [comment]
Been heavily rethinking our own relay efforts since it's less of a requirement with Defender in place. Couple of givens and a conclusion:
* We already have a robust authentication system in place as part of our OAuth2 OIDC provider. 
* Since recently we have been properly signing an authentication request message for both Metamask and Torus key wallets upon signup and also on signin if no address is known yet. Because of this we can assume that the address stored in our accounts collection in the database is the address belonging to the account requesting the API. 
* Authenticating relayed contract calls currently seems redundant since calls to the contracts are only allowed to be made from the  single whitelisted account being the static address of the Defender Relayer (formerly the API admin account). 
* Since we made signup a requirement (and adding an email address optional for wallets) I see no reason why we can not forward all users through the signup process in auth. This allows all users to do authenticated API requests and we can in turn use the AccountProxy in the API controllers to retrieve the address of the account (instead of recovering the address from a signed contract call).
* Our contracts still do _msgSender() calls in various methods to recover the signer of the contract call. Those were designed to be public contract methods that could be signed by an unknown wallet and relayed through our own RelayHubFacet allowing unauthenticated users to interact with our system. As we made signup a requirement, this is no longer necessary.
Conclusion:
* Adding an extra function argument address _sender to the methods depending on _msgSender() would make our current relaying efforts redundant and allows us to remove the RelayHubFacet completely. Minimising attack surface and code complexity.

Made progress on this, which results in transaction signing from the wallet no longer being required.
